packages/compass-workspaces/src/index.ts

```diff
@@ -88,12 +91,38 @@ export function activateWorkspacePlugin(
 
   addCleanup(cleanupLocalAppRegistries);
 
-  const setupInstanceListeners = (instance: MongoDBInstance) => {
+  const setupInstanceListeners = (
+    connectionId: string,
+    instance: MongoDBInstance
+  ) => {
     on(instance, 'change:collections._id', (collection: Collection) => {
       const { _id: from } = collection.previousAttributes();
       store.dispatch(collectionRenamed(from, collection.ns));
     });
 
+    on(instance, 'change:databases.is_non_existent', (database: Database) => {
+      const namespaceId = `${connectionId}.${database._id}`;
+      const databaseInfo = {
+        isNonExistent: database.is_non_existent,
+      };
+      store.dispatch(updateDatabaseInfo(namespaceId, databaseInfo));
+    });
+
+    on(
+      instance,
+      'change:collections.is_non_existent',
+      (collection: Collection) => {
+        const namespaceId = `${connectionId}.${collection._id}`;
+        const collectionInfo = {
+          isTimeSeries: collection.isTimeSeries,
+          isReadonly: collection.readonly ?? collection.isView,
+          sourceName: collection.sourceName,
```

Comment:
For case like this I'd probably prefer to either have a dedicated action or subscribe to all fields you're updating, otherwise you're listening here to one thing, but updating others, it's a but confusing to navigate. Taking into account we're going to refactor this, that's fine for now

Reply:
great point. with refactor in mind, i'll merge this for now!